### PR TITLE
Authorization maximization celebration

### PR DESCRIPTION
**Pull request by jayshenk**
Please review. Clone the auth branch and run `rails db:migrate`. You will need to log in after registering. Enjoy your baguette. We like satisfied customers. Looking forward to your feedback.

### CODE REVIEW

**Comment by vedala** [comment]
I attempted all actions on the application as per requirements. Everything working as expected.

- User register, login and logout working good.
- Great job with flash messages. Saw messages for actions along happy path as well as invalid actions (e.g. trying to view someone else's todo item).

File: app/models/todo.rb [hunk 1]
```diff
@@ -1,4 +1,7 @@
 class Todo < ActiveRecord::Base
+  belongs_to :user
+  belongs_to :category, optional: true
+
```
Comment (preeV42500) [inline app/models/todo.rb:4]: Nice use of the optional: true to indicate that having a category for a todo is optional

**Comment by p-reznick** [comment]
I second Kalyan's review, satisfies all of the requirements and flash messages are well implemented and comprehensive.

File: config/routes.rb [hunk 2]
```diff
@@ -1,6 +1,17 @@
...
+  post "/login", to: "sessions#create"
+  post "/logout", to: "sessions#destroy"
+
+  get "/register", to: "users#new", as: :new_user
+  post "/register", to: "users#create", as: :users
+
   resources :todos
+  resources :users, only: [:show, :edit, :update]
+  resources :categories, only: [:show, :new, :create]
 end
```
Comment (preeV42500) [inline config/routes.rb:17]: Our team made categories a nested resource under users, perhaps making things more complicated. Good job with making the categories routes work without having them nested under users!

**Pull request merged**